cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -280,9 +283,67 @@ void DecisionTreeBase<T, L>::plant(
 
   total_temp_mem = tempmem->totalmem;
   MLCommon::TimerCPU timer;
-  grow_deep_tree(data, labels, rowids, n_sampled_rows, ncols,
-                 tree_params.max_features, dinfo.NLocalrows, sparsetree, treeid,
-                 tempmem);
+  bool use_experimental_backend = false;
+
+  if (tree_params.use_experimental_backend == true) {
+    use_experimental_backend = true;
+
+    if (tree_params.split_algo != SPLIT_ALGO::GLOBAL_QUANTILE) {
+      CUML_LOG_WARN(
+        "Experimental backend does not yet support histogram split algorithm");
+      CUML_LOG_WARN(
+        "To use experimental backend set split_algo = 1 (GLOBAL_QUANTILE)");
+      use_experimental_backend = false;
+    }
+
+    if (tree_params.max_depth <= 0 || tree_params.max_depth >= 14) {
+      CUML_LOG_WARN(
+        "Experimental backend does not yet support arbitrary depth trees");
+      CUML_LOG_WARN("To use experimental backend set 0 < max_depth < 14");
+      use_experimental_backend = false;
+    }
+
+    if (tree_params.max_features != 1.0) {
+      CUML_LOG_WARN(
+        "Experimental backend does not yet support feature sub-sampling");
+      CUML_LOG_WARN("To use experimental backend set max_features = 1.0");
+      use_experimental_backend = false;
+    }
+
+    if (tree_params.quantile_per_tree == false) {
+      CUML_LOG_WARN(
+        "Experimental backend does not yet support per tree quantile "
```

Comment:
This condition is backwards - should be quantile_per_true == true that disables

Reply:
Nice catch!